### PR TITLE
First Look

### PR DESCRIPTION
**Pull request by philipmassouh**
I made some changes/improvements/suggestions here. I am going to comment more notes

### CODE REVIEW

File: cviche.py [hunk 1]
```diff
@@ -2,11 +2,18 @@
 import utilities
 import sys
 
+from pathlib import Path
+from enum import Enum, auto
+
+class Modes(Enum):
+   exp = auto()
+   edu = auto()
+   skl = auto()
```
Comment (philipmassouh) [inline cviche.py]: enums are nice, can use `is` check instead of `==` when comparing to them.

File: cviche.py [hunk 2]
```diff
@@ -2,11 +2,18 @@
...
 # import json
 # print(json.dumps(master)) #can dump into a json file later
 
 
-def experience(template, csv, tag):
+def experience(template: Path, csv: Path, tag) -> str:
     master = utilities.csv_to_table(csv)
```
Comment (philipmassouh) [inline cviche.py:12]: I recommend `pandas` with `read_csv`

File: cviche.py [hunk 3]
```diff
@@ -21,8 +28,7 @@ def skills(filename, tag):
     master = utilities.csv_to_table(filename)
     return utilities.select_values_with_tag(master, "skill", "tags", tag)
```
Comment (philipmassouh) [inline cviche.py:24]: symbolic constants

File: cviche.py [hunk 4]
```diff
@@ -21,8 +28,7 @@ def skills(filename, tag):
     master = utilities.csv_to_table(filename)
```
Comment (philipmassouh) [inline cviche.py]: filename can be a `Path` object: `from pathlib import Path`

File: utilities.py [hunk 5]
```diff
@@ -1,21 +1,20 @@
...
-
+def csv_to_table(filename: Path):
     with open(filename, "r") as fin:
         csv_reader = csv.DictReader(fin, delimiter=",")
         table = list(csv_reader)
     return table
 
 
-def replace_placeholders(document, replacements):
+def replace_placeholders(document: str, replacements: dict[str, str]) -> str:
```
Comment (philipmassouh) [inline utilities.py:17]: typehints

**Review by philipmassouh — COMMENTED** [review]
lgtm bro

File: utilities.py (hunk 5 above)
Comment (rudyghill) [reply utilities.py:17]: I like this and need this

File: cviche.py (hunk 1 above)
Comment (rudyghill) [reply cviche.py]: I'm going to drop this because I have plans to make cviche more general like we talked about

File: cviche.py (hunk 4 above)
Comment (rudyghill) [reply cviche.py]: I'll bring in these changes, but I need to make sure I'm using path right such that my program doesn't crash

**Review by rudyghill — CHANGES_REQUESTED** [review]
I like what I am seeing but can you discard the enum? And can you verify the program won't crash by using the Path types?

And for now, I don't think I will use pandas

**Review by rudyghill — APPROVED** [review]
I removed the enums myself

**Pull request merged**